packages/react/src/date-picker/DatePicker.test.tsx

```diff
@@ -281,6 +281,68 @@ describe('DatePicker', () => {
 		expect(await getInput()).toHaveValue('5 Jun 2023');
 	});
 
+	describe('allowedDateFormats', () => {
+		const onChange = jest.fn();
+		const onInputChange = jest.fn();
+
+		it('formats when a valid format is entered', async () => {
+			renderDatePicker({
+				allowedDateFormats: ['dd-MM-yyyy'],
+				label: 'Example',
+				onChange,
+				onInputChange,
+			});
+
+			const dateString = '23-05-2023';
+			const date = parseDate(dateString) as Date;
+
+			// Type a valid date in the input field that is in the allowed format
+			await userEvent.type(await getInput(), dateString);
+			await userEvent.keyboard('{Tab}');
+
+			expect(onChange).toHaveBeenLastCalledWith(date);
+			expect(onInputChange).not.toHaveBeenCalled();
```

Comment:
These are implementation details. If we want to assert that `'formats when a valid format is entered'` we should get the input value after the blur event.

Reply:
Just wrote better tests 🙇‍♂️